script/bigscreenplayer.js

```diff
@@ -82,58 +82,77 @@ define('bigscreenplayer/bigscreenplayer',
-            }
+          initialPlaybackTimeEpoch = bigscreenPlayerData.initialPlaybackTime;
+          // overwrite initialPlaybackTime with video time (it comes in as epoch time for a sliding/growing window)
+          bigscreenPlayerData.initialPlaybackTime = SlidingWindowUtils.convertToSeekableVideoTime(bigscreenPlayerData.initialPlaybackTime, windowStartTime);
+        }
 
-            initialPlaybackTimeEpoch = bigscreenPlayerData.initialPlaybackTime;
+        mediaKind = bigscreenPlayerData.media.kind;
+        endOfStream = windowType !== WindowTypes.STATIC && (!bigscreenPlayerData.initialPlaybackTime && bigscreenPlayerData.initialPlaybackTime !== 0);
 
-            // overwrite initialPlaybackTime with video time (it comes in as epoch time for a sliding/growing window)
-            bigscreenPlayerData.initialPlaybackTime = SlidingWindowUtils.convertToSeekableVideoTime(bigscreenPlayerData.initialPlaybackTime, windowStartTime);
-          }
+        playerComponent = new PlayerComponent(
+          playbackElement,
+          bigscreenPlayerData,
+          windowType,
+          enableSubtitles,
+          mediaStateUpdateCallback,
+          device
+        );
 
-          mediaKind = bigscreenPlayerData.media.kind;
+        successCallback();
 
-          liveSupport = newLiveSupport;
-          windowType = newWindowType;
-          endOfStream = windowType !== WindowTypes.STATIC && (!bigscreenPlayerData.initialPlaybackTime && bigscreenPlayerData.initialPlaybackTime !== 0);
+        var availableCdns = bigscreenPlayerData.media.urls.map(function (media) {
+          return media.cdn;
+        });
 
-          playerComponent = new PlayerComponent(
-            playbackElement,
-            bigscreenPlayerData,
-            windowType,
-            enableSubtitles,
-            mediaStateUpdateCallback,
-            device,
-            liveSupport
-          );
+        DebugTool.keyValue({key: 'available cdns', value: availableCdns});
+        DebugTool.keyValue({key: 'current cdn', value: bigscreenPlayerData.media.urls[0].cdn});
+        DebugTool.keyValue({key: 'url', value: bigscreenPlayerData.media.urls[0].url});
+      }
 
-          var availableCdns = bigscreenPlayerData.media.urls.map(function (media) {
-            return media.cdn;
-          });
+      return {
+        init: function (playbackElement, bigscreenPlayerData, newWindowType, enableSubtitles, newDevice, callbacks) {
+          Chronicle.init();
+          device = newDevice;
+          windowType = newWindowType;
+          serverDate = bigscreenPlayerData.serverDate;
 
-          DebugTool.keyValue({key: 'available cdns', value: availableCdns});
-          DebugTool.keyValue({key: 'current cdn', value: bigscreenPlayerData.media.urls[0].cdn});
-          DebugTool.keyValue({key: 'url', value: bigscreenPlayerData.media.urls[0].url});
+          if (needToGetManifest() && !bigscreenPlayerData.time) {
+            ManifestLoader().load(
```

Comment:
Should this be a function or reference a specific instance? Rather than instantiating it everytime